### ISSUE TITLE
Clearing the clipboard seems random

### ISSUE THREAD

**Issue by ihatemakinganaccount** [hashemi-hossein/memory-guardian#17]
The app seems to clear the clipboard in random time intervals, between 30 seconds and 3 minutes, regardless of how the timer is set (tried up to 600 secs). Only when the timer is set to very short (under 30 secs), the settings seem to be more or less respected.

The clipboard also clears every time the app is launched - not sure if that is intended, but it's unexpected.

Additionally, the "small popup" (I assume that means toast) doesn't ever show up. Toasts work fine in other apps.

Notifications work, both the persistent one and the notification about the clipboard being cleared.

Also I get the clear notification even when the clipboard is empty, i.e. it just pops up randomly every so often...

Also also, the "get current clipboard content" doesn't do anything (might be an Android limitation as they've been restricting clipboard access lately).

OS: CalyxOS 5.5.1 (based on AOSP - Android 14)

Thanks for your effort!

**Comment by ihatemakinganaccount** [comment]
Am I the only one with this issue?

**Comment by poyribn** [comment]
yes, the clipboard seems to be cleared on a timer that resets every time the timer reaches zero, which ignores clipboard changes.

that unfortunately means there's always a chance the clipboard is cleared instantly when i'm unlucky enough that the timer already almost reached the number of the interval setting.

the timer _should_ reset only when the clipboard contents change.

until that is fixed, Memory&nbsp;Guardian is unusable.
